borsh-schema-derive-internal/src/enum_schema.rs

```diff
@@ -96,6 +96,7 @@ pub fn process_enum(input: &ItemEnum, cratename: Ident) -> syn::Result<TokenStre
             }
         }
         anonymous_defs.extend(quote! {
+            #[allow(dead_code)]
```

Comment:
@dj8yfo I don't have all the necessary context here, so I would ask you to look into it: do we even need those unused fields in the struct? I somehow feel uneasy about silencing the compiler here as it might indicate that something is missing / not implemented when it should be.

Reply:
I've looked into it.
The fields with their types and possibly `#[borsh_skip]` attributes are needed to derive correct inner impl `BorshSchema` for inner struct inside of method of `BorshSchema` impl for enum. So that type info is still needed.
The compiler is complaining that these structs never get constructed and their fields read. 
Which is kind of normal, provided that the structs are [invisible](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1c3059950fbd7fc81a32611ba27eec97) outside of the [method's scope](https://github.com/near/borsh-rs/blob/master/borsh-schema-derive-internal/src/enum_schema.rs#L155-L173). 
They're accessed by their types instead : `<InnerStruct>::declaration()`, `<InnerStruct as borsh::BorshSchema>::add_definitions_recursively(definitions)`.

Reply:
Thanks for looking into it!